docs/porting/custom-target-porting.md

```diff
@@ -0,0 +1,259 @@
+
+   All the other configurations for the board are inherited from the MCU Family configuration called `FAMILY_STM32`.  
+
+   Here are some other typical changes that might be needed.
+   
+   - (Optional) You can also use `device_has_add` to add additional drivers.
+
+   <span class="notes">**Note:** If you choose to add a driver, you may have to provide the driver implementation if it is not already available for your hardware.</span>
+
+   - (Optional) Similarly, you can use `features_add`, `features_remove`, `components_add`, `components_remove`, `macros_add` and `macros_remove` to add/remove configurations.
+
+## Configuring the target code directories
+
+In some cases, the target source code directories follow a similar structure as the target configuration, but they could have a few more levels.
+
+For example, in the `mbed-os/targets` folder, the target directories for DISCO_L475VG_IOT01A follow this pattern:
+
+   ```
+   mbed-os
+       |_targets
+             |_TARGET_STM            <- MCU VENDOR
+             |      |_TARGET_STM32L4             <- MCU FAMILY
+             |             |_TARGET_STM32L475xG               <- MCU
+             |                    |_TARGET_DISCO_L475VG_IOT01A     <- Board
+   ```
+
+Boards typically inherit files that support the MCU, MCU family and MCU vendor. When adding a new board, you need to add a new set of files for the board.
+
+You may be wondering why there are more directory levels than target configuration levels. This is because many targets use the `extra_labels_add` feature in the target configuration. The following keywords `STM32L4`, `STM32L475xG`, `STM32L475VG` resolve to `TARGET_STM32L4`, `TARGET_STM32L475xG`, `TARGET_STM32L475VG` respectively. With those labels applied, these directory names get included in the build for this target.
+
+1. Create a new directory called `TARGET_IMAGINARYBOARD` at the top level of your project.
+   
+   Your directory should look something like this:
+   
+   ```
+   custom_target.json
+   TARGET_IMAGINARYBOARD
+   mbed-os
+   .mbed
+   mbed_settings.py
+   mbed-os.lib
+   ```
+   
+1. Now place source files for the board.
+   
+   Inspect the files at `mbed-os\targets\TARGET_STM\TARGET_STM32L4\TARGET_STM32L475xG\TARGET_DISCO_L475VG_IOT01A`. You should find the following files or similar:
+   
+   `PeripheralNames.h`, `PeripheralPins.c`, `PinNames.h`, `system_clock.c`
+   
+   Copy the files into your new `TARGET_IMAGINARYBOARD` directory.
+   
+   The files provide these capabilities.
+      
+   - `PeripheralNames.h` describes the available peripherals and their base addresses.
+   - `PeripheralPins.c` describes the available pins and their association with peripherals.
+   - `PinNames.h` sets macros for pins that define their function.
+   - `system_clock.c` vendor specific file that initializes the system and sets up the clocks.
+      
+1. Modify the files.
```

Comment:
Again, maybe have all the previous steps as a "getting ready" bit and this as a new header, since it's the meat of the issue